src/App.tsx

```diff
@@ -1,18 +1,51 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
+const total = 42;
+const items = getNumbers(1, total)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
+
+  const indexOfLastItem = useMemo(() => {
+    const index = currentPage * perPage;
+
+    if (index > total) {
+      return total;
+    }
+
+    return index;
+  }, [currentPage, perPage, total]);
+
+  const indexOfFirstItem = useMemo(() => {
+    return (currentPage - 1) * perPage;
+  }, [perPage, currentPage]);
+
+  const itemsPerPage = useMemo(() => (
+    items.slice(indexOfFirstItem, indexOfLastItem)),
+  [indexOfFirstItem, indexOfLastItem]);
+
+  const handleChangeSelect = useCallback(
+    (e: React.ChangeEvent<HTMLSelectElement>) => {
+      setPerPage(+e.target.value);
+      setCurrentPage(1);
+    }, [setPerPage, setCurrentPage],
+  );
+
+  const handleChangePage = useCallback((page: number) => {
+    setCurrentPage(page);
+  }, [setCurrentPage]);
```

Comment:
Memoization is a costly thing for memory. Using it for every component rendering might decrease performance since it stores results that increase the memory used for the program. We should only use memoization when there is a clear benefit.
You don't really need to use it for simple values or functions. If you’re interested in it, you can read more about it [here](https://kentcdodds.com/blog/usememo-and-usecallback), for example